File: solutions-by-language/cpp/2785_sort_vowels_in_a_string.cpp
class Solution {
public:
    string sortVowels(string s) {
        unordered_set<char> vowels = {'a','e','i','o','u','A','E','I','O','U'};
        vector <char> vowels_chars;

        for (char c : s) {
            if(vowels.count(c)) {
                vowels_chars.push_back(c);
            }
        }

        sort(vowels_chars.begin(), vowels_chars.end());

        int idx = 0;
        for(int i = 0 ; i < s.size(); i++) {
            if(vowels.count(s[i])) {
                s[i] = vowels_chars[idx++];
            }
        }
        return s;
    }
};

File: solutions-by-language/python/2785_sort_vowels_in_a_string.py
class Solution:
    def sortVowels(self, s: str) -> str:
        vowels = set("aeiouAEIOU")
        vowel_chars = [c for c in s if c in vowels]
        vowel_chars.sort()

        result = []
        vowel_index = 0

        for c in s:
            if c in vowels:
                result.append(vowel_chars[vowel_index])
                vowel_index += 1
            else:
                result.append(c)

        return ''.join(result)